ipmikvm

```diff
@@ -33,10 +33,38 @@
 # - JViewer-unknown-version that comes with Quanta BMC
 #
 # See also: ipmiview
-#
-test -z "$HOME" && echo 'missing $HOME' && exit 1
+if test -z "$APP_CACHE_DIR"; then
+    if ! test -z "$HOME" && test -d "$HOME/.local/lib/ipmikvm"; then
+        APP_CACHE_DIR="$HOME/.local/lib/ipmikvm"
+    else
+        APP_CACHE_DIR="${XDG_CACHE_HOME:-${HOME?}/.cache}/ipmikvm"
+    fi
+fi
 set -u
-APP_CACHE_DIR="$HOME/.local/lib/ipmikvm"
+
+if ! command -v unpack200 >/dev/null; then
+    JAVA_HOME="${JAVA_HOME:-/usr/lib/jvm/java-8-openjdk}"
+    if ! test -d "$JAVA_HOME"; then
+        echo "$JAVA_HOME and unpack200 is missing" >/dev/stderr
+        exit 1
+    else
+        unpack200() {
+            "$JAVA_HOME/bin/unpack200" "$@"
+        }
+    fi
+fi
+
+for binary in java unzip curl awk base64 grep sed flock; do
+    if ! command -v "$binary" >/dev/null; then
+        echo "'$binary' not found" >/dev/stderr
+        exit 1
+    fi
+done
+
+mkdir -p "$APP_CACHE_DIR"
+
+LOCKFD=4
+exec 4<>$APP_CACHE_DIR/lock
```

Comment:
This should be $LOCKFD.

Reply:
That sadly doesn't work 😥;

```
./ipmikvm: line 67: exec: 4: not found
```

I _could_ use `eval exec "$LOCKFD"'<>"$APP_CACHE_DIR/lock"'` but I found that very weird and hard to understand, especially since the `exec` is so close to the `LOCKFD` variable